app/gui2/src/components/GraphEditor.vue

```diff
@@ -318,42 +314,16 @@ const creatingNodeFromButton: Interaction = {
 
 const creatingNodeFromPortDoubleClick: Interaction = {
   init: () => {
-    componentBrowserInputContent.value = ''
+    // component browser usage is set in event handler
     componentBrowserVisible.value = true
   },
 }
 
-async function handleFileDrop(event: DragEvent) {
-  // A vertical gap between created nodes when multiple files were dropped together.
-  const MULTIPLE_FILES_GAP = 50
-
-  if (!event.dataTransfer?.items) return
-  ;[...event.dataTransfer.items].forEach(async (item, index) => {
-    try {
-      if (item.kind === 'file') {
-        const file = item.getAsFile()
-        if (!file) return
-        const clientPos = new Vec2(event.clientX, event.clientY)
-        const offset = new Vec2(0, index * -MULTIPLE_FILES_GAP)
-        const pos = graphNavigator.clientToScenePos(clientPos).add(offset)
-        const uploader = await Uploader.Create(
-          projectStore.lsRpcConnection,
-          projectStore.dataConnection,
-          projectStore.contentRoots,
-          projectStore.awareness,
-          file,
-          pos,
-          projectStore.isOnLocalBackend,
-          event.shiftKey,
-          projectStore.executionContext.getStackTop(),
-        )
-        const uploadResult = await uploader.upload()
-        graphStore.createNode(pos, uploadedExpression(uploadResult))
-      }
-    } catch (err) {
-      console.error(`Uploading file failed. ${err}`)
-    }
-  })
+const creatingNodeFromEdgeDrop: Interaction = {
+  init: () => {
+    // component browser usage is set in event handler
+    componentBrowserVisible.value = true
```

Comment:
Should this interaction also close the component browser when it is canceled? E.g. when an edge drag starts.

Reply:
This canceling/hiding routine is a bit of spaghetti. I tried to simplify it a bit (at least settle who is responsible for actual hiding CB)